docs/model_developers/modules.rst

```diff
@@ -122,3 +122,60 @@ The format for requires in requirements.txt is the folllowing:
  * Or a repository location such as  git+https://github.com/project/python-foo The correct syntax
    to use is then: eggname@git+https://../repository#branch with branch being optional.
 
+
+Working on modules
+==================
+Since modules often have dependencies on other modules, it is common to develop against multiple
+modules (or a project and one or more modules) simultaneously. One might for example need to
+extend a dependency module to add support for some new feature. Because this use case is so common,
+this section will describe how to work on multiple modules simultaneously so that any changes are
+visible the compiler. This procedure is of course applicable for working on a single module as well.
+
+Setting up the dev environment
+------------------------------
+v1 modules
+^^^^^^^^^^
+A v1 module is a module that is not packaged and published to a pip index but is solely distributed
+via git. V1 modules are installed on the fly by the compiler and can be found in the project's
+`libs` directory. Any changes you make to the module source in the `libs` dir will be reflected in
+the next compile. If you need to be able to make changes to multiple modules simultaneously, the
+simplest way to set up your development environment is to create a project that imports the desired
+modules, then run a compile against it. Alternatively, you can manually clone the module repos.
+
+v2 modules
+^^^^^^^^^^
+A v2 module in development form has mostly the same structure as a v1 module. The main difference is
+that it is meant to be published as a Python package. A project or another v2 module then lists all
+required v2 modules as dependencies in its `pyproject.toml`. The compiler does not install v2
+modules on the fly. In line with how Python depdencies work in general they are expected to be
+installed in advance. As a result they will not be placed in the project's `libs` directory.
+
+To set up your development environment, first clone all modules you wish to develop against and
```

Comment:
Shouldn't we simply provide a command like `inmanta modules install` that installs everything?
And e.g. `inmanta modules dev-install` that also installs all dev dependencies? 

Or is this in a context outside of a specific project? 

also, it this installation in a normal venv or the compiler venv?

Reply:
- this is either in the context of a project or the context of a single module. I think we will indeed need something like `inmanta module install` and `inmanta project install` if we need to wrap `pip install -e`. For an editable install of a dependency I think we'll always need to add manual steps though.
- I was thinking of the development env outside of the compiler venv. I'd see these dependencies as something managed by the model developer (he can choose to do editable installs for example), so I don't think it would be right to put it in the compiler venv.

Reply:
this command exists and is called `inmanta module install` (but that installs in the compiler venv and doesn't install dev dependencies).